File: quiz_app_flutter/lib/domain/usecase/quiz/getquizdata_bysetandtopic_usecase.dart
import 'package:quiz_app_flutter/core/resource.dart';
import 'package:quiz_app_flutter/domain/entities/quiz/quiz_data.dart';
import 'package:quiz_app_flutter/domain/repository/quiz_repository.dart';

class GetQuizDataBySetAndTopicUseCase {
  final QuizRepository repository;

  GetQuizDataBySetAndTopicUseCase(this.repository);

  Stream<Resource<List<QuizData>>> call(String fileName) {
    return repository.getQuizzesBySetAndTopic(fileName);
  }
}

File: quiz_app_kmp/composeApp/src/commonMain/kotlin/com/tnm/quizappkmp/domain/usecase/quiz/GetQuizDataBySetAndTopicUseCase.kt
package com.tnm.quizappkmp.domain.usecase.quiz

import com.tnm.quizappkmp.domain.model.Resource
import com.tnm.quizappkmp.domain.model.quiz.QuizData
import com.tnm.quizappkmp.domain.repositry.QuizRepository
import kotlinx.coroutines.flow.Flow

class GetQuizDataBySetAndTopicUseCase(
    private val repository: QuizRepository
) {
    operator fun invoke(fileName: String): Flow<Resource<List<QuizData>>> {
        return repository.getQuizzesBySetAndTopic(fileName)
    }
}